wordpress/wp-content/themes/mts_ecommerce/options/demo-importer/demo-files/demo-1/theme_options.txt
###a:127:{s:8:"last_tab";s:1:"0";s:8:"mts_logo";s:79:"https://demo.mythemeshop.com/ecommerce-demo3/files/2015/07/logo-transparent.png";s:11:"mts_favicon";s:70:"https://demo.mythemeshop.com/ecommerce-demo4/files/2015/07/favicon.png";s:14:"mts_touch_icon";s:0:"";s:14:"mts_metro_icon";s:0:"";s:20:"mts_twitter_username";s:0:"";s:14:"mts_feedburner";s:0:"";s:15:"mts_header_code";s:0:"";s:18:"mts_analytics_code";s:0:"";s:14:"mts_copyrights";s:74:"Theme by <a href="https://mythemeshop.com/" rel="nofollow">MyThemeShop</a>";s:14:"mts_responsive";s:1:"1";s:7:"mts_rtl";s:1:"0";s:15:"mts_prefetching";s:1:"0";s:13:"mts_lazy_load";s:1:"0";s:20:"mts_lazy_load_thumbs";s:1:"0";s:21:"mts_lazy_load_content";s:1:"0";s:12:"mts_async_js";s:1:"1";s:21:"mts_remove_ver_params";s:1:"1";s:15:"mts_optimize_wc";s:1:"0";s:16:"mts_color_scheme";s:7:"#4b5f6b";s:17:"mts_color_scheme2";s:7:"#7dce82";s:17:"mts_color_scheme3";s:7:"#e55934";s:10:"mts_layout";s:8:"sclayout";s:19:"mts_payment_section";s:1:"1";s:11:"mts_payment";a:3:{i:0;a:6:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"1";s:5:"title";s:15:"Secure Payments";s:4:"icon";s:4:"lock";s:11:"description";s:69:"Lorem ipsum dolor sit amet. Morbi at nisl lorem, vel porttitor justo.";s:3:"url";s:0:"";}i:1;a:6:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"2";s:5:"title";s:20:"Money Back Guarantee";s:4:"icon";s:3:"usd";s:11:"description";s:69:"Lorem ipsum dolor sit amet. Morbi at nisl lorem, vel porttitor justo.";s:3:"url";s:0:"";}i:2;a:6:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"3";s:5:"title";s:13:"Free Delivery";s:4:"icon";s:5:"truck";s:11:"description";s:66:"Pellentesque habitant morbi netus tristique senectus et malesuada.";s:3:"url";s:0:"";}}s:14:"mts_top_footer";s:1:"1";s:34:"mts_accepted_payment_method_images";a:10:{i:0;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"5";s:24:"mts_payment_method_title";s:6:"PayPal";s:24:"mts_payment_method_image";s:6:"paypal";s:31:"mts_payment_method_custom_image";s:0:"";}i:1;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"6";s:24:"mts_payment_method_title";s:11:"TwoCheckout";s:24:"mts_payment_method_image";s:3:"2co";s:31:"mts_payment_method_custom_image";s:0:"";}i:2;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"7";s:24:"mts_payment_method_title";s:16:"American Express";s:24:"mts_payment_method_image";s:4:"amex";s:31:"mts_payment_method_custom_image";s:0:"";}i:3;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"8";s:24:"mts_payment_method_title";s:6:"Cirrus";s:24:"mts_payment_method_image";s:6:"cirrus";s:31:"mts_payment_method_custom_image";s:0:"";}i:4;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"9";s:24:"mts_payment_method_title";s:5:"Delta";s:24:"mts_payment_method_image";s:5:"delta";s:31:"mts_payment_method_custom_image";s:0:"";}i:5;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"10";s:24:"mts_payment_method_title";s:8:"Discover";s:24:"mts_payment_method_image";s:8:"discover";s:31:"mts_payment_method_custom_image";s:0:"";}i:6;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"11";s:24:"mts_payment_method_title";s:10:"MasterCard";s:24:"mts_payment_method_image";s:10:"mastercard";s:31:"mts_payment_method_custom_image";s:0:"";}i:7;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"12";s:24:"mts_payment_method_title";s:12:"MoneyBookers";s:24:"mts_payment_method_image";s:12:"moneybookers";s:31:"mts_payment_method_custom_image";s:0:"";}i:8;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"13";s:24:"mts_payment_method_title";s:4:"VISA";s:24:"mts_payment_method_image";s:4:"visa";s:31:"mts_payment_method_custom_image";s:0:"";}i:9;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"14";s:24:"mts_payment_method_title";s:6:"Switch";s:24:"mts_payment_method_image";s:6:"switch";s:31:"mts_payment_method_custom_image";s:0:"";}}s:23:"mts_footer_social_icons";a:6:{i:0;a:7:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"16";s:28:"mts_footer_social_icon_title";s:8:"Facebook";s:22:"mts_footer_social_icon";s:8:"facebook";s:26:"mts_footer_social_icon_url";s:1:"#";s:28:"mts_footer_social_icon_color";s:7:"#ffffff";s:31:"mts_footer_social_icon_bg_color";s:7:"#3b5998";}i:1;a:7:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"17";s:28:"mts_footer_social_icon_title";s:7:"Twitter";s:22:"mts_footer_social_icon";s:7:"twitter";s:26:"mts_footer_social_icon_url";s:1:"#";s:28:"mts_footer_social_icon_color";s:0:"";s:31:"mts_footer_social_icon_bg_color";s:7:"#00aced";}i:2;a:7:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"18";s:28:"mts_footer_social_icon_title";s:8:"LinkedIn";s:22:"mts_footer_social_icon";s:8:"linkedin";s:26:"mts_footer_social_icon_url";s:1:"#";s:28:"mts_footer_social_icon_color";s:0:"";s:31:"mts_footer_social_icon_bg_color";s:7:"#007bb6";}i:3;a:7:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"19";s:28:"mts_footer_social_icon_title";s:11:"Google Plus";s:22:"mts_footer_social_icon";s:11:"google-plus";s:26:"mts_footer_social_icon_url";s:1:"#";s:28:"mts_footer_social_icon_color";s:0:"";s:31:"mts_footer_social_icon_bg_color";s:7:"#dd4b39";}i:4;a:7:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"20";s:28:"mts_footer_social_icon_title";s:6:"Flickr";s:22:"mts_footer_social_icon";s:6:"flickr";s:26:"mts_footer_social_icon_url";s:1:"#";s:28:"mts_footer_social_icon_color";s:0:"";s:31:"mts_footer_social_icon_bg_color";s:7:"#ff0084";}i:5;a:7:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"21";s:28:"mts_footer_social_icon_title";s:9:"Pinterest";s:22:"mts_footer_social_icon";s:11:"pinterest-p";s:26:"mts_footer_social_icon_url";s:1:"#";s:28:"mts_footer_social_icon_color";s:0:"";s:31:"mts_footer_social_icon_bg_color";s:7:"#cb2027";}}s:16:"mts_first_footer";s:1:"1";s:20:"mts_first_footer_num";s:1:"4";s:14:"mts_background";a:9:{s:5:"color";s:7:"#fafcfd";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}}s:21:"mts_footer_background";a:9:{s:5:"color";s:7:"#344655";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}}s:14:"mts_custom_css";s:0:"";s:12:"mts_lightbox";s:1:"0";s:17:"mts_header_layout";s:1:"6";s:14:"mts_sticky_nav";s:1:"0";s:20:"mts_show_primary_nav";s:1:"1";s:23:"mts_primary_nav_bgcolor";s:7:"#f4f7f9";s:22:"mts_show_secondary_nav";s:1:"1";s:19:"mts_header_section2";s:1:"1";s:17:"mts_header_search";s:1:"1";s:32:"mts_header_bottom_ad_nav_1_title";s:12:"Save 18% Now";s:31:"mts_header_bottom_ad_nav_1_desc";s:20:"Coupon Code: ec02142";s:32:"mts_header_bottom_ad_nav_2_title";s:18:"DISCOUNT UP TO 25%";s:31:"mts_header_bottom_ad_nav_2_desc";s:40:"Buy more save more. caupon code: ec02142";s:32:"mts_header_bottom_ad_nav_3_title";s:18:"GET ADDITIONAL 10%";s:31:"mts_header_bottom_ad_nav_3_desc";s:39:"FOR NEW CUSTOMERS. caupon code: ec02142";s:21:"mts_header_background";a:9:{s:5:"color";s:7:"#fafcfd";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}}s:19:"mts_featured_slider";s:1:"1";s:17:"mts_custom_slider";a:2:{i:0;a:9:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"23";s:23:"mts_custom_slider_title";s:9:"Style Up!";s:26:"mts_custom_slider_subtitle";s:30:"With our amazing collection of";s:25:"mts_custom_slider_heading";s:9:"Swim Suit";s:28:"mts_custom_slider_subheading";s:35:"Order Now. <span>Get 35% off</span>";s:29:"mts_custom_slider_button_text";s:14:"Start Shopping";s:23:"mts_custom_slider_image";s:3:"999";s:22:"mts_custom_slider_link";s:50:"https://demo.mythemeshop.com/ecommerce-demo4/shop/";}i:1;a:9:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"24";s:23:"mts_custom_slider_title";s:9:"Check Out";s:26:"mts_custom_slider_subtitle";s:25:"Our amazing collection of";s:25:"mts_custom_slider_heading";s:7:"Cameras";s:28:"mts_custom_slider_subheading";s:35:"Order Now. <span>Get 40% off</span>";s:29:"mts_custom_slider_button_text";s:14:"Start Shopping";s:23:"mts_custom_slider_image";s:3:"996";s:22:"mts_custom_slider_link";s:50:"https://demo.mythemeshop.com/ecommerce-demo4/shop/";}}s:15:"mts_home_layout";a:2:{s:7:"enabled";a:7:{s:7:"banners";s:7:"Banners";s:8:"lookbook";s:8:"Lookbook";s:12:"product_cats";s:25:"Product Categories Slider";s:12:"product_tabs";s:12:"Product Tabs";s:12:"product_grid";s:13:"Products Grid";s:6:"offers";s:6:"Offers";s:6:"brands";s:6:"Brands";}s:8:"disabled";a:3:{s:5:"posts";s:12:"Latest Posts";s:6:"banner";s:11:"Thin Banner";s:7:"welcome";s:7:"Welcome";}}s:16:"lookbook_heading";s:8:"Lookbook";s:15:"lookbook_images";a:7:{i:0;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"26";s:5:"title";s:0:"";s:5:"image";s:0:"";}i:1;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"27";s:5:"title";s:0:"";s:5:"image";s:3:"952";}i:2;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"28";s:5:"title";s:0:"";s:5:"image";s:3:"953";}i:3;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"29";s:5:"title";s:0:"";s:5:"image";s:3:"954";}i:4;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"30";s:5:"title";s:0:"";s:5:"image";s:3:"950";}i:5;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"31";s:5:"title";s:0:"";s:5:"image";s:3:"953";}i:6;a:4:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"32";s:5:"title";s:0:"";s:5:"image";s:3:"954";}}s:14:"browse_heading";s:21:"Browse our Categories";s:21:"mts_tabs_hover_effect";s:4:"lexi";s:13:"mts_home_tabs";a:1:{s:7:"enabled";a:3:{s:16:"best_sellers_tab";s:12:"Best sellers";s:16:"new_products_tab";s:12:"New Arrivals";s:13:"top_rated_tab";s:9:"Top Rated";}}s:20:"product_grid_heading";s:0:"";s:16:"product_grid_num";s:1:"6";s:20:"product_grid_img_num";s:1:"3";s:20:"products_grid_sortby";s:6:"recent";s:14:"offers_heading";s:14:"Limited Offers";s:13:"offers_sortby";s:6:"recent";s:14:"brands_heading";s:13:"Shop By Brand";s:12:"brand_images";a:5:{i:0;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"34";s:5:"title";s:0:"";s:5:"image";s:3:"955";s:4:"link";s:1:"#";}i:1;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"35";s:5:"title";s:0:"";s:5:"image";s:3:"956";s:4:"link";s:1:"#";}i:2;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"36";s:5:"title";s:0:"";s:5:"image";s:3:"957";s:4:"link";s:1:"#";}i:3;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"37";s:5:"title";s:0:"";s:5:"image";s:3:"958";s:4:"link";s:1:"#";}i:4;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:2:"38";s:5:"title";s:0:"";s:5:"image";s:3:"959";s:4

wordpress/wp-content/themes/mts_ecommerce/options/js/options.js
var changeWarn = false;
jQuery(document).ready(function($){
	var History = window.History;
	function loadTabs(){
		jQuery('.nhp-opts-group-tab').hide();
		jQuery('.nhp-opts-group-tab-link-li').removeClass('active');
		if(jQuery('#last_tab').val() == ''){
			jQuery('.nhp-opts-group-tab:first').slideDown('fast');
			jQuery('#nhp-opts-group-menu li:first').addClass('active');
		}else{
			tabid = jQuery('#last_tab').val();
			jQuery('#'+tabid+'_section_group').slideDown('fast');
			jQuery('#'+tabid+'_section_group_li').addClass('active');
			if (jQuery('#'+tabid+'_section_group_li').closest('#nhp-opts-homepage-accordion').length)
				jQuery('#nhp-opts-homepage-accordion').show();
			if (jQuery('#'+tabid+'_section_group_li').closest('#nhp-opts-blog-accordion').length)
				jQuery('#nhp-opts-blog-accordion').show();
		}
	}
	jQuery('.nhp-opts-group-tab-link-a').click(function(){
		clear_search();

		if (this.id == 'accordion_section_group_li_a') {
		   jQuery('#nhp-opts-homepage-accordion').slideToggle();
		   return false;
		}
		if (this.id == 'accordion_section_group_li_a_2') {
		   jQuery('#nhp-opts-blog-accordion').slideToggle();
		   return false;
		}
		var $this = jQuery(this);
		if ($this.parent().hasClass('active'))
			return false;
		var relid = jQuery(this).attr('data-rel');
		
		jQuery('.nhp-opts-group-tab').hide();
		jQuery('#'+relid+'_section_group').fadeIn(400);

		jQuery('.nhp-opts-group-tab-link-li').removeClass('active');
		$this.parent().addClass('active');

		//jQuery("html, body").animate({ scrollTop: jQuery('#nhp-opts-header').offset().top - 48 }, 500);

		History.pushState( {tab:relid}, document.title, "themes.php?page=theme_options&tab="+relid );

		return false;
	});

	History.Adapter.bind( window,'load statechange', function( event ){
		var State = History.getState();
		var tab = State.data.tab;

		if ( typeof tab !== 'undefined' ) {
			jQuery('#'+tab+'_section_group_li_a').click();
		} else {
			loadTabs();
		}
	});
	jQuery('#nhp-opts-form-wrapper').submit(function(){
		var currentTabID = jQuery(this).find('#nhp-opts-group-menu li.active a').attr('data-rel');
		jQuery('#last_tab').val(currentTabID);
	});
	

	if(jQuery('#nhp-opts-save').is(':visible')){
		jQuery('#nhp-opts-save').delay(4000).slideUp('slow');
	}
	function addTypographyData(container, form) {
		var i = 0;
		container.find(".collections .collection").each(function() {
			form.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][preview_text]', jQuery(this).find(".preview_text").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][preview_color]', jQuery(this).find(".preview_color li.current a").attr("class"))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_family]', jQuery(this).find(".font_family").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_variant]', jQuery(this).find(".font_variant").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_size]', jQuery(this).find(".font_size").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_color]', jQuery(this).find(".font_color").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][css_selectors]', jQuery(this).find(".css_selectors").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][additional_css]', jQuery(this).find(".additional_css").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][backup_font]', jQuery(this).find(".backup_font").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][default]', jQuery(this).attr("data-default"))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][collection_title]', jQuery(this).find(".collection_title").val());
			
			i++;
		});
	}
	
	if(jQuery('#nhp-opts-imported').is(':visible')){
		jQuery('#nhp-opts-imported').delay(4000).slideUp('slow');
	}
	
	jQuery('#nhp-opts-footer').find('#savechanges').click(function(e) {
		// add typography data via hidden fields before submitting
		if (typography_isloaded) {
			addTypographyData(jQuery('#google_typography'),jQuery('#nhp-opts-form-wrapper'));
		}
		// AJAX save
		/*
		jQuery('#savechanges').prop('disabled', true).after('<div class="spinner" id="ajax-saving"></div>');
		changeWarn = false;
		var $form = jQuery('#nhp-opts-form-wrapper');
		jQuery.post( $form.attr('action'), $form.serialize() ).done(function() {
			jQuery('#ajax-saving').remove();
			jQuery('#savechanges').prop('disabled', false).after('<div id="ajax-saved">Settings saved!</div>');
			setTimeout(function() { jQuery('#ajax-saved').fadeOut('slow', function() { jQuery('#ajax-saved').remove(); }); }, 2000);
		});
		return false;
		*/
		// AJAX save end
	});

	jQuery('input, textarea, select').change(function(){
		if (!changeWarn) {
		  //jQuery('#nhp-opts-save-warn').slideDown('slow');
		  changeWarn = true;
		}
	});

	jQuery('#nhp-opts-form-wrapper').submit(function() {
		changeWarn = false;
	});

	window.onbeforeunload = confirmExit;
	function confirmExit() {
		if (changeWarn) {
			return nhpopts.leave_page_confirm;
		}
	}
	
	jQuery('#nhp-opts-import-code-button').click(function(e){
		e.preventDefault();
		jQuery('#nhp-opts-import-code-wrapper').toggle().find('#import-code-value').val('');
	});

	jQuery('#nhp-opts-export-code-copy').click(function(e){
		e.preventDefault();
		jQuery('#nhp-opts-export-code').toggle().select();
	});
	
	// Presets
	function scrollImportLogToBottom(){
		var element = document.getElementById("importing-modal-content");
		element.scrollTop = element.scrollHeight;
	}
	jQuery('#presets .preset .import-demo-button, #presets .preset .import-demo-widgets-button, #presets .preset .import-demo-options-button').on('click', function(e) {
		e.preventDefault();

		var $this = jQuery(this);
		var $parent = $this.closest('.preset');
		var confirmText = nhpopts.import_opt_confirm;
		if ( $this.hasClass('import-demo-button') ) {
			confirmText = nhpopts.import_all_confirm;
		}
		if ( $this.hasClass('import-demo-widgets-button') ) {
			confirmText = nhpopts.import_widget_confirm;
		}

		var result = confirm( confirmText );
		if ( result ) {

			var data = {};
			data.action = "mts_install_demo";
			data.demo_import_id = $parent.attr("data-demo-id");
			data.nonce = $parent.attr("data-nonce");
			data.demo_import_options = '1';
			data.demo_import_content = '0';
			data.demo_import_widgets = '0';

			if ( $this.hasClass('import-demo-button') ) {
				data.demo_import_content = '1';
				data.demo_import_widgets = '1';
			}

			if ( $this.hasClass('import-demo-widgets-button') ) {
				data.demo_import_widgets = '1';
			}

			$this.magnificPopup({
				items: {
					src: '#importing-overlay',
					type: 'inline'
				},
				modal: true
			}).magnificPopup('open');

			var last_response_len = false;
			jQuery.ajax( ajaxurl, {
				data: data,
				xhrFields: {
					onprogress: function(e) {

						var this_response, response = e.currentTarget.response;
						if(last_response_len === false) {

							this_response = response;
							last_response_len = response.length;

						} else {

							this_response = response.substring(last_response_len);
							last_response_len = response.length;
						}

						jQuery('#importing-modal-content').append(this_response);
						scrollImportLogToBottom();
					}
				}
			})
			.done(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.import_done);
				jQuery('#importing-modal-footer-info').text(nhpopts.import_done);
				jQuery('#importing-modal-footer-button').show();
			})
			.fail(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.import_fail);
				jQuery('#importing-modal-footer-info').text(nhpopts.import_fail);
				jQuery('#importing-modal-footer-button').show();
			});
		}

		return false;
	});

	jQuery('.remove-demo-button').on('click', function(e) {
		e.preventDefault();

		var result = confirm( nhpopts.remove_all_confirm );
		if ( result ) {

			var $this = jQuery(this);
			var data = {};
			data.action = "mts_install_demo";
			data.mts_remove_demos = "1";
			data.nonce = $this.attr("data-nonce");

			$this.magnificPopup({
				items: {
					src: '#importing-overlay',
					type: 'inline'
				},
				modal: true
			}).magnificPopup('open');

			var last_response_len = false;
			jQuery.ajax( ajaxurl, {
				data: data,
				xhrFields: {
					onprogress: function(e) {

						var this_response, response = e.currentTarget.response;
						if(last_response_len === false) {

							this_response = response;
							last_response_len = response.length;

						} else {

							this_response = response.substring(last_response_len);
							last_response_len = response.length;
						}

						jQuery('#importing-modal-content').append(this_response);
						scrollImportLogToBottom();
					}
				}
			})
			.done(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.remove_done);
				jQuery('#importing-modal-footer-info').text(nhpopts.remove_done);
				jQuery('#importing-modal-footer-button').show();
			})
			.fail(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.remove_fail);
				jQuery('#importing-modal-footer-info').text(nhpopts.remove_fail);
				jQuery('#importing-modal-footer-button').show();
			});
		}

		return false;
	});

	function mtsRemoveURLParameter(url, parameter) {

		//prefer to use l.search if you have a location/link object
		var urlparts= url.split('?');
		if ( urlparts.length >= 2 ) {

			var prefix= encodeURIComponent(parameter)+'=';
			var pars= urlparts[1].split(/[&;]/g);

			//reverse iteration as may be destructive
			for (var i= pars.length; i-- > 0;) {
				//idiom for string.startsWith
				if (pars[i].lastIndexOf(prefix, 0) !== -1) {
					pars.splice(i, 1);
				}
			}

			url= urlparts[0]+'?'+pars.join('&');
			return url;

		} else {

			return url;
		}
	}
	
	jQuery('#importing-modal-footer-button').on('click', function(e) {
		e.preventDefault();
		jQuery(this).prop('disabled', true ).text(nhpopts.reloading_page);
		var a = mtsRemoveURLParameter( window.location.href , 'tab' );
		window.location.href = a;
	});

	
	// Confirm import
	jQuery('#nhp-opts-import').click(function() {
		return confirm(nhpopts.import_confirm);
	});
	
	// Confirm reset
	jQuery('input[name="'+nhpopts.opt_name+'[defaults]"]').click(function() {
		return confirm(nhpopts.reset_confirm);
	});
	
	// Disallow submission by enter key
	jQuery('#nhp-opts-form-wrapper').find('input').keydown(function(event){
		if ( event.keyCode == 13 ){
			event.preventDefault();
		}
	});
	
	// Floating footer
	var $footer = jQuery('#nhp-opts-footer');
	var $bottom = jQuery('#nhp-opts-bottom');
	
	
	$footer.addClass('floating');
	jQuery(document).on('scroll', function(){
		if ($bottom.isOnScreen()) {
			$footer.removeClass('floating');
		} else {
			$footer.addClass('floating');
		}
	});
	if ($bottom.isOnScreen()) {
		$footer.removeClass('floating');
	}
	
	// Needs JS sizing when position:fixed
	var footer_padding = $footer.innerWidth() - $footer.width();
	function resizeFloatingElements() {
		var w = jQuery('#nhp-opts-form-wrapper').width();
		$footer.width(w - footer_padding);
	}
	resizeFloatingElements();
	
	var resizeTimer;
	jQuery(window).resize(function() {
		clearTimeout(resizeTimer);
		resizeTimer = setTimeout(resizeFloatingElements, 100);
	});

	// Child theme creator
	jQuery('#nhp-opts-child-button').on('click', function(e) {
		e.preventDefault();
		var elem		 = jQuery(this),
			childName	= jQuery('#nhp-opts-child-name'),
			childNameVal = childName.val();

		if ( '' === childNameVal ) {
			alert( nhpopts.child_theme_name_empty );
		} else {
			jQuery.ajax({
				url: ajaxurl, 
				method: 'post',
				data: {
                    'action': 'mts_child_theme',
                    'child_name': childNameVal,
                    '_ajax_nonce': $('#mts_child_theme_nonce').val()
                },
				beforeSend: function() {
					elem.prop('disabled', true);
					childName.prop('disabled', true);
				},
				success: function(data) {
					elem.prop('disabled', false);
					childName.val('').prop('disabled', false);
					mtsUpdateChildThemesList();
				}
			});
		}
	});
	// Refresh the existing child themes list
    function mtsUpdateChildThemesList() {
        jQuery.post( ajaxurl, { action: 'mts_list_child_themes', '_ajax_nonce': $('#mts_child_theme_nonce').val() }, function( response ) {
            if ( response ) {
                jQuery('#child-theme-list-wrap').html( response );
            }
        });
    }
	
	$('#search-theme-options').keydown(function(e) {
		if (e.keyCode == 27) {
			clear_search( true );
			$(this).val('').blur();
			return false;
		}
	}).keyup(function (e) {
		var value = $(this).val();
		if ( value == '' ) {
			clear_search( true );
			return false;
		}
		search_options( value );
	}).blur(function(event) {
		var value = $(this).val();
		if ( value == '' ) {
			$(this).closest('div').removeClass('active');
		}
	});
	$('.options-search-link').click(function(event) {
		event.preventDefault();
		if ($(this).closest('div').hasClass('active')) {
			var query = $(this).closest('div').addClass('active').find('input').val();
			search_options( query );
		}
		$(this).closest('div').addClass('active').find('input').focus();
	});
	$('.theme-options-clear-search').click(function(event) {
		event.preventDefault();
		clear_search( true );
		$(this).siblings('input').val('').trigger('blur');
	});
	$('#nhp-opts-header .docsupport').click(function(event){
		event.preventDefault();
		$('.nhp-opts-group-tab-link-a[data-rel=support]').trigger('click');
	});

	// Reverse jQuery plugin
	jQuery.fn.reverse = [].reverse;

	var searched = false;
	function search_options( query ) {
		searched = true;
		// de-select tabs
		$('.nhp-opts-group-tab-link-li.active').removeClass('active');
		$('.nhp-opts-info-field').hide();
		query = query.toLowerCase();
		var noresults = true;
		$('#nhp-opts-main').children('div').hide().each(function(index, el) {
			if (this.id == 'typography_default_section_group') {
				return true;
			}
			var $contents = $(this);

			//var $title = $contents.find('h2');
			//var $desc = $contents.find('.nhp-opts-section-desc');
			
			$contents.children('table').children('tbody').children('tr').each(function(index, el) {
				var $row = $(this);
				$row.hide();
				if ($row.find('th').text().toLowerCase().indexOf( query ) !== -1) {
					$contents.show();
					$row.show();
					noresults = false;
				}

			}).reverse().each(function(index, el) {
				if ($(this).is(':visible') && $(this).find('.buttonset-hide').length) {
					$(this).find('#nhp-opts-button-show-below').each(function(event){
						if (jQuery(this).hasClass('ui-state-active')) {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').show();
						} else {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').hide();
						}
					});
				}
			});
		});
		if (noresults) {
			$('#options-search-no-results').show();
		}
	}

	function clear_search( clicktab ) {
		if ( ! searched ) {
			return false;
		}
		$('#nhp-opts-main').children().each(function(index, el) {
			$(this).find('tr').show().reverse().each(function(index, el) {
				if ($(this).find('.buttonset-hide').length) {
					$(this).find('#nhp-opts-button-show-below').each(function(event){
						if (jQuery(this).hasClass('ui-state-active')) {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').show();
						} else {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').hide();
						}
					});
				}
			});
		});
		$('.nhp-opts-info-field').show();
		if ( clicktab ) {
			$('#nhp-opts-main').children().hide().first().show();
			$('#0_section_group_li').addClass('active');
		} else {
			$('#search-theme-options').val('').parent().removeClass('active');
		}
		searched = false;
	}
});

jQuery(window).on('load',function() {
	jQuery('#savechanges').prop('disabled', false);
});

jQuery.fn.isOnScreen = function(){

	var win = jQuery(window);

	var viewport = {
		top : win.scrollTop(),
		left : win.scrollLeft()
	};
	viewport.right = viewport.left + win.width();
	viewport.bottom = viewport.top + win.height();

	var bounds = this.offset();
	bounds.right = bounds.left + this.outerWidth();
	bounds.bottom = bounds.top + this.outerHeight();

	return (!(viewport.right < bounds.left || viewport.left > bounds.right || viewport.bottom < bounds.top || viewport.top > bounds.bottom));

};

jQuery.fn.addHiddenField = function(name, value) {
	this.each(function () {
		var elem_id = name.replace(/\W/g, '-').replace(/--+/g, '-').replace(/(^-|-$)/, '');
		if (jQuery('#'+elem_id).length) {
			// elem exists, change value
			jQuery('#'+elem_id).val(value);
		} else {
			// elem doesn't exist, create
			var input = jQuery("<input>").attr("type", "hidden").attr("id", elem_id).attr("name", name).val(value);
			jQuery(this).append(jQuery(input));
		}
		
	});
	return this;
};

var fnDelay = (function(){
  var timer = 0;
  return function(callback, ms){
	clearTimeout (timer);
	timer = setTimeout(callback, ms);
  };
})();
